### DISCUSSION TITLE
How is the image resource for PipelineOutput created?

### DISCUSSION THREAD

**Discussion by ohmaya** [o3de/o3de#17421] [Help Wanted]
MainPipeline.pass:

```
            "PassData": {
                "$type": "PassData",
                "PipelineGlobalConnections": [
                    {
                        "GlobalName": "PipelineOutput",
                        "Slot": "PipelineOutput"
                    }
                ]
            },
```

PipelineOutput is referred to in DepthPrePass request:
```
                        {
                            "LocalSlot": "PipelineOutput",
                            "AttachmentRef": {
                                "Pass": "PipelineGlobal",
                                "Attachment": "PipelineOutput"
                            }
                        }
```
And, **where is this PipelineGlobal pass defined**? Might be a SwapChainPass?

Thanks.

**Comment by VickyAtAZ** [comment]
"PipelineGlobal" is a special keyword but not a particular pass.
If "PipelineGlobal" is specified, it would look for a the pass connection reference in render pipeline's global connections. 
Any pass may add "PipelineGlobalConnections" to their PassData to create a render pipeline global connection from its pass connection. 

For example, in o3de\Gems\Atom\Feature\Common\Assets\Passes\DepthParent.pass, the DepthParentTemplate adds 3 pipeline global connections in its PassData. This means, when other pass uses 
```
"AttachmentRef": {
                                "Pass": "PipelineGlobal",
                                "Attachment": "DepthMSAA"
                            }
```
to reference a pass connection/attachment, it would use DepthParentTemplate's DepthMSAA.

**Reply by ohmaya** [reply]
Yeah, "const Name PipelineGlobalKeyword{"PipelineGlobal"};". It was just an incidental question for the main question. So, it means the image resource for PipelineOutput is not created in PipelineGlobal pass.

I will think the image resource is created along with the swapchain, then only the attachment is created in SwapChainPass:
```
m_swapChainAttachment = aznew PassAttachment();
pipelineOutput->SetAttachment(m_swapChainAttachment);
```

https://github.com/aws-lumberyard-dev/o3de/commit/bb0959a775c684d19896ce33e6bd5c91d967b20e#diff-4d68c889c10324f59061914491e1864abfe2533ab3bc248e78a96f5c1a5d6593
This FSR2 integration adds a RenderOutput slot along with PipelineOutput. However, **it seems the image for RenderOutput is not created, so if the depth pass refers to RenderOutput, the created depth buffer is of (1, 1, 1)**.
